src/main/java/edu/tseidler/model/Board.java

```diff
@@ -75,4 +87,50 @@ public Fields getFields() {
     public boolean ifFull() {
         return maxCol * maxCol == fields.getTakenFieldsNumber();
     }
+
+    public boolean doWeHaveAWinner() {
+        Coordinates lastCoords = fields.getLastCoords();
+        BoardField lastMark = fields.getLastMark();
+
+        return (checkGeneralizedWinner(lastCoords, lastMark, 1, 0)) ||
+                checkGeneralizedWinner(lastCoords, lastMark, 0, 1) ||
+                checkGeneralizedWinner(lastCoords, lastMark, 1, -1) ||
+                checkGeneralizedWinner(lastCoords, lastMark, 1, 1);
+    }
+
+
+    private boolean checkGeneralizedWinner(Coordinates lastCoords, BoardField lastMark, int rowInc, int colInc) {
```

Comment:
This method is too long and too complicated, try to extract some code to new methods (with descriptive names) so the code will be more readable